packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts

```diff
@@ -0,0 +1,279 @@
+          if (ds.groupedOptions) {
+            dictionaryConditions.logs.push(
+              eventFilterToDictionaryCondition(filter, ds.groupedOptions),
+            );
+          } else if (ds.options?.address || filter.topics) {
+            dictionaryConditions.logs.push(
+              eventFilterToDictionaryCondition(filter, ds.options),
+            );
+          } else {
+            // do nothing;
+          }
+          break;
+        }
+        default:
+      }
+    }
+  }
+
+  if (!dictionaryConditions.logs.length) {
+    delete dictionaryConditions.logs;
+  } else {
+    dictionaryConditions.logs = uniqBy(dictionaryConditions.logs, isEqual);
+  }
+
+  if (!dictionaryConditions.transactions.length) {
+    delete dictionaryConditions.transactions;
+  } else {
+    dictionaryConditions.transactions = uniqBy(
+      dictionaryConditions.transactions,
+      isEqual,
+    );
+  }
+  return dictionaryConditions;
+}
+
+export class EthDictionaryV2 extends DictionaryV2<
+  EthereumBlock,
+  SubqlDatasource,
+  EthDictionaryV2QueryEntry
+> {
+  constructor(
+    endpoint: string,
+    nodeConfig: NodeConfig,
+    eventEmitter: EventEmitter2,
+    project: SubqueryProject,
+    chainId?: string,
+  ) {
+    super(
+      endpoint,
+      chainId ?? project.network.chainId,
+      nodeConfig,
+      eventEmitter,
+    );
+  }
+
+  static async create(
+    endpoint: string,
+    nodeConfig: NodeConfig,
+    eventEmitter: EventEmitter2,
+    project: SubqueryProject,
+    chainId?: string,
+  ): Promise<EthDictionaryV2> {
+    const dictionary = new EthDictionaryV2(
+      endpoint,
+      nodeConfig,
+      eventEmitter,
+      project,
+      chainId,
+    );
+    await dictionary.init();
+    return dictionary;
+  }
+
+  buildDictionaryQueryEntries(
+    dataSources: SubqlDatasource[],
+  ): EthDictionaryV2QueryEntry {
+    const filteredDs = ethFilterDs(dataSources);
+    return buildDictionaryV2QueryEntry(filteredDs);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async getData(
+    startBlock: number,
+    endBlock: number,
+    limit: number = MIN_FETCH_LIMIT,
+  ): Promise<DictionaryResponse<IBlock<EthereumBlock> | number> | undefined> {
+    return super.getData(startBlock, endBlock, limit, {
+      blockHeader: true,
+      logs: { transaction: true },
+      transactions: { log: true },
+    });
+  }
+
+  convertResponseBlocks<RFB = RawEthBlock>(
+    data: RawDictionaryResponseData<RFB>,
+  ): DictionaryResponse<IBlock<EthereumBlock>> | undefined {
+    try {
+      const blocks: IBlock<EthereumBlock>[] = (
+        (data.blocks as RawEthBlock[]) || []
+      ).map(rawBlockToEthBlock);
+      return {
+        batchBlocks: blocks,
+        lastBufferedHeight: blocks.length
+          ? blocks[blocks.length - 1].block.number
+          : undefined,
+      };
+    } catch (e) {
+      logger.error(
+        e,
+        `Failed to handle block response ${JSON.stringify(data)}`,
```

Comment:
This could be a large amount of data to log

Reply:
I removed `${JSON.stringify(data)`, I think from error msg from `rawBlockToEthBlock` is very detailed should be sufficent